cmd/crank/render/runtime_docker.go

```diff
@@ -261,29 +263,88 @@ func (r *RuntimeDocker) createContainer(ctx context.Context, cli *client.Client)
 	// Find a random, available port. There's a chance of a race here, where
 	// something else binds to the port before we start our container.
 
-	lis, err := net.Listen("tcp", "localhost:0")
+	// Check if we're using a remote Docker daemon
+	// Remote means TCP or SSH connection, not Unix socket (which is always local)
+	dockerHost := os.Getenv("DOCKER_HOST")
+	isRemoteDocker := dockerHost != "" && !strings.Contains(dockerHost, "unix://")
+
+	// Determine the bind address
+	bindAddr := "localhost"
+	if isRemoteDocker {
+		// When using remote Docker, bind to all interfaces
+		bindAddr = "0.0.0.0"
+	}
+
+	lis, err := net.Listen("tcp", fmt.Sprintf("%s:0", bindAddr))
 	if err != nil {
 		return "", "", errors.Wrap(err, "cannot get available TCP port")
 	}
 
-	containerAddr := lis.Addr().String()
+	listenerAddr := lis.Addr().String()
 	_ = lis.Close()
 
-	spec := fmt.Sprintf("%s:9443/tcp", containerAddr)
-
-	expose, bind, err := nat.ParsePortSpecs([]string{spec})
-	if err != nil {
-		return "", "", errors.Wrapf(err, "cannot parse Docker port spec %q", spec)
+	// Determine the address that containers should connect to
+	var containerConnectAddr string
+	if isRemoteDocker {
+		// Check for explicit host override
+		if renderHost := os.Getenv("CROSSPLANE_RENDER_HOST"); renderHost != "" {
```

Comment:
I'm having a hard time wrapping my brain around this. The PR description refers to it as the "callback address", and it's named `CROSSPLANE_RENDER_HOST`. This makes it sound like the function (running in Docker) would try to connect back to this address.

However if I follow the code correctly, it looks like this is actually the address `crank render` is going to try to use to connect (via gRPC) to the function in the container, right? So I think it's actually overriding how `crossplane render` will call the function (in Docker). Essentially forcing it to call a specific address, not one derived from `DOCKER_HOST`?